astrobee/survey_manager/survey_planner/tools/survey_planner/command_astrobee

```diff
@@ -0,0 +1,512 @@
+            current_robot = ""
+        sim = True
+
+    ns = ""
+    # If we're commanding a robot remotely
+    if current_robot != robot_name:
+        rospy.loginfo("We're commanding a namespaced robot!")
+        ns = " -remote -ns " + robot_name
+        # Command executor will add namespace for bridge forwarding
+        command_executor = CommandExecutor(robot_name)
+    else:
+        command_executor = CommandExecutor("")
+    process_executor = ProcessExecutor(robot_name)
+
+    # Initialize exit code
+    exit_code = 0
+
+    if goal == "dock":
+        exit_code += process_executor.send_command_recursive("rosrun executive teleop_tool -dock" + ns + " -berth " + config_static["berth"][to_bay])
+
+    elif goal == "undock":
+        exit_code += process_executor.send_command_recursive("rosrun executive teleop_tool -undock" + ns)
+
+    elif goal == "move":
+        exit_code += process_executor.send_command_recursive("rosrun executive teleop_tool -move " + config_static["bays_move"][to_bay] + ns)
+        # Change exposure if needed
+        exposure_value = exposure_change(config_static, from_bay, to_bay)
+        if exposure_value != 0:
+            exit_code += command_executor.change_exposure(exposure_value)
+        # Change map if needed
+        map_name = map_change(config_static, from_bay, to_bay)
+        if map_name != "":
+            exit_code += command_executor.change_map(map_name)
+
+    elif goal == "panorama":
+        exit_code += command_executor.start_recording("pano_" + to_bay + "_" + run)
+        exit_code += process_executor.send_command_recursive("rosrun inspection inspection_tool -panorama -panorama_mode '5_mapper_and_hugin' " + config_static["bays_move"][to_bay] + ns)
+        exit_code += command_executor.stop_recording()
+
+    elif goal == "stereo":
+        exit_code += command_executor.start_recording("stereo_" + bay + "_" + run)
+        exit_code += command_executor.set_plan("plans/ISAAC/" + get_stereo_traj(from_bay, to_bay))
+        exit_code += command_executor.run_plan()
+        exit_code += command_executor.stop_recording()
+
+    return exit_code
+
+
+class CustomFormatter(argparse.ArgumentDefaultsHelpFormatter):
+    pass
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description=__doc__, formatter_class=CustomFormatter
+    )
+
+    parser.add_argument(
+        "robot_name",
+        default="",
+        help="Robot name executing the command.",
+    )
+    parser.add_argument(
+        "command_name",
+        help="Prefix for bagfiles to merge. Bags should all be in the current working directory.",
+    )
+    parser.add_argument(
+        "to_bay",
+        default="",
+        help="Target bay destination.",
+    )
+    parser.add_argument(
+        "from_bay",
+        default="",
+        help="Target bay start.",
+    )
+    parser.add_argument(
+        "run",
+        default="",
+        help="Run number, increases as we add attempts.",
+    )
```

Comment:
It seems like `argparse` named positional arguments are not a great fit here, since different commands take arguments with different names. I would switch the interface to receiving a PDDL-formatted action in a single argument and using `plan_interpreter.py` to parse that into a dict with correct field names.

Reply:
fixed